modules/diskq/qdisk.c

```diff
@@ -296,22 +296,23 @@ _maybe_truncate_file(QDisk *self, gint64 expected_size)
 
 #if !SYSLOG_NG_HAVE_POSIX_FALLOCATE
 static gint
-_compat_preallocate(QDisk *self, gint64 size)
+_compat_preallocate(int fd, off_t offset, off_t len)
 {
   const size_t buf_size = QDISK_RESERVED_SPACE;
-  gint64 buf_write_iterations = size / buf_size;
-  gint64 additional_write_size = size - buf_write_iterations * buf_size;
+  gint64 buf_write_iterations = (len - offset) / buf_size;
```

Comment:
`len` is not necessarily greater than `offset`.

I think this calculation is incorrect.

> The function posix_fallocate() ensures that disk space is
       allocated for the file referred to by the file descriptor fd for
       the bytes in the range starting at offset and continuing for len
       bytes.

Based on this, we should write exactly `len` bytes after `offset`.

Reply:
Thanks, fixed.